featuretools/entityset/entityset.py

```diff
@@ -671,6 +671,9 @@ def add_dataframe(self,
         self._add_references_to_metadata(dataframe)
         return self
 
+    def __setitem__(self, key, value):
+        self.add_dataframe(dataframe=value, dataframe_name=key)
```

Comment:
Would it be worth handling the situation where a user passes in a woodwork-initialized dataframe that doesn't have a `df.ww.name` set (or if the `df.ww.name` is different from `key`) differently than `add_dataframe` does? 

Currently the first situation would end up with an error `Cannot add a Woodwork DataFrame to EntitySet without a name` and the second raises a warning that the `dataframe_name` parameter is going to be ignored in favor of the name at `df.ww.name`. 

I know there was some discussion with @thehomebrewnerd  and @rwedge when we decided on that behavior for `add_dataframe` in the first place, but somehow `setitem` feels different to me. I know in woodwork's setitem, we just use the key value for a series' name even if we have to overwrite an already-existing name, and I think it might be work updating the woodwork name on the dataframe before calling `add_dataframe`.

Reply:
@tamargrey: When looking at the behavior of `add_dataframe` when using a Woodwork initialized dataframe without a name, I do find it strange to get an error. I would think `add_dataframe`, when given a name, would add that name to the Woodwork dataframe. Maybe that behavior should change? I feel pretty strongly that `__setitem__` should be a very thin layer over `add_dataframe` and call it with its default values, and let `add_dataframe` handle errors, warnings, etc.

Reply:
Actually, a partially initialized Woodwork Dataframe causes all the issues that are described in the Note text, so maybe that is enough in this case

Reply:
I think the first case should be handled by having `add_dataframe` add a name to woodwork-initialized dataframes that lack a name.  There's a larger issue #1740 discussing it

Reply:
Maybe @gsheni can chime in here, but I believe the underlying sprite of this ticket is to improve DX with Featuretools. So if I have an EntitySet and I just want to quickly add a dataframe (that hasn't been WW-initialized), I can just do `es['dataframe'] = df`. If the user wants more granular control, they can do that by using `add_dataframe` and additionally initializing the df using WW before adding it.

Reply:
It'd be nice for this to also work with already woodworked dataframes but I'm fine with those edge cases being handled in other tickets

Reply:
Yes, let's go for a good initial pass and then we can improve on it later on.